pom.xml

```diff
@@ -140,24 +145,25 @@
             <version>2.0</version>
         </dependency>
 
-        <dependency><!-- exists in the core -->
+        <dependency>
             <groupId>org.jenkins-ci.modules</groupId>
             <artifactId>instance-identity</artifactId>
-            <version>1.3</version>
-            <scope>provided</scope>
+            <version>2.1</version><!-- matches 2.60.3 -->
         </dependency>
 
-        <!--TEST DEPS-->
-        <dependency> <!-- from org.eclipse.jetty:jetty-server:9.2.15.v20160210 -->
+        <!-- Required for java8 -->
+        <dependency> <!-- from org.eclipse.jetty:jetty-server:9.4.5.v20170502-->
             <groupId>javax.servlet</groupId>
             <artifactId>javax.servlet-api</artifactId>
-            <scope>test</scope>
+            <version>3.1.0</version>
```

Comment:
same, should probably be configured to take from the Core instead of pinned here

Reply:
This should be resolved, please verify

Reply:
Surely this could just be removed?

Reply:
Unfortunately, the tests fail if it isn't included. 


```
       <a href="https://jenkins.io/redirect/users-mailing-list">The users list</a> might be also useful in understanding what has happened.</p><h2>Stack trace</h2><pre style="margin:2em; clear:both">java.lang.NoSuchMethodError: javax.servlet.http.HttpServletResponse.getStatus()I
	at org.eclipse.jetty.server.handler.ErrorHandler.doError(ErrorHandler.java:142)
	at org.eclipse.jetty.server.handler.ErrorHandler.handle(ErrorHandler.java:78)
	at org.eclipse.jetty.server.Response.sendError(Response.java:655)
	at javax.servlet.http.HttpServletResponseWrapper.sendError(HttpServletResponseWrapper.java:118)
	at javax.servlet.http.HttpServletResponseWrapper.sendError(HttpServl
```

Reply:
Then someone did something wrong here. Track down what. It is not normal to depend explicitly on `servlet-api` from a plugin POM.

Reply:
Just so I am clear too, you think the entire dependency should be able to be removed, not simply removing the scope/test and replacing with the version pin, right?

Reply:
Probably yes. Try deleting the explicit dep and then checking `mvn dependency:tree` to see if something else is improperly pulling in this dep. I saw something about Jetty in there? Might suffice to just have another `<exclude>` somewhere.

Reply:
I was able to remove the pinning, change the scope from `test` => `provided` and this does seem to be working...I'm pretty confused, as I think I tried this before and was not able to get it to work, but perhaps there was some additional pinning that I fixed that makes this work.

It looks like there are some servlet hooks here: https://github.com/jenkinsci/github-plugin/blob/497f43251e3bfc130f125f75e45f73e06894361d/src/main/java/com/cloudbees/jenkins/GitHubWebHookCrumbExclusion.java that is requiring the servlet extension, I'm still trying to figure out exactly what the purpose of it is.

Reply:
It is normal enough for plugins to be referring to Servlet API classes—they are in the signature of several APIs defined in Jenkins core, and generally by [Stapler](http://stapler.kohsuke.org/reference.html). However the dep is normally defined in the parent POM; it should not need to be mentioned explicitly in the plugin’s POM.